core/src/main/java/com/amplifyframework/storage/options/StorageUploadFileOptions.java

```diff
@@ -127,15 +105,19 @@ public static StorageUploadFileOptions defaultInstance() {
         private String contentType;
         private Map<String, String> metadata;
 
+        private Builder() {
+            this.metadata = new HashMap<>();
+        }
```

Comment:
You think it'd be better to have an empty map vs. a null if there is no metadata?

Reply:
IIRC it was already like this, but I removed it in one of my recent PRs. This is just a re-addition. Besides, `TransferUtility` simply calls overloaded method with a `new ObjectMetadata()` upon calling upload without metadata anyways.

Reply:
Ok sounds good